mmeval/metrics/rouge.py

```diff
@@ -0,0 +1,307 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+# This class is modified from `torchmetrics
+# <https://github.com/Lightning-AI/metrics/blob/master/src/torchmetrics/text/rouge.py>`_.
+import re
+from collections import Counter
+from difflib import SequenceMatcher
+from typing import (TYPE_CHECKING, Any, Callable, Dict, List, Optional,
+                    Sequence, Tuple, Union)
+
+from mmeval import BaseMetric
+from mmeval.metrics.utils import get_tokenizer, inference_language
+from mmeval.utils import try_import
+
+if TYPE_CHECKING:
+    import nltk
+else:
+    nltk = try_import('nltk')
+
+
+def _get_results(matches: int, pred_len: int,
+                 reference_len: int) -> Dict[str, float]:
+    """This computes precision, recall and F1 score based on matches.
+
+    Args:
+        matches (int): A number of matches or a length of
+            the longest common subsequence.
+        pred_len (int): A length of a tokenized predicted sentence.
+        reference_len (int): A length of a tokenized referenced sentence.
+    Returns:
+        Dict[str, float]: Calculate f1 value, recall rate and accuracy rate.
+    """
+    precision = matches / pred_len
+    recall = matches / reference_len
+    if precision == recall == 0.0:
+        return dict(
+            precision=float(0.0), recall=float(0.0), fmeasure=float(0.0))
+
+    fmeasure = 2 * precision * recall / (precision + recall)
+    return dict(
+        precision=float(precision),
+        recall=float(recall),
+        fmeasure=float(fmeasure))
+
+
+def _rougeL_score(pred: Sequence[str],
+                  reference: Sequence[str]) -> Dict[str, float]:
+    """This computes precision, recall and F1 score for the Rouge-L metric.
+
+    Args:
+        pred (Sequence[str]): A predicted sentence.
+        reference (Sequence[str]): A referenced sentence.
```

Comment:
```suggestion
        reference (Sequence[str]): A referenced sentence.

```

Reply:
It seems that there is no change between the modified code and the original code?

Reply:
A blank line is necessary before `Returns`~